app/users/user.service.js

```diff
@@ -1,24 +1,24 @@
 const UserResult = Object.freeze({ Success: 0, UserExists: 1 })
-
-const users = []
+const UserDAO = require('./user.db')
 
 const UserService = {
   createUser,
   userCount,
   UserResult
 }
 
-function createUser (userName) {
-  if (users.includes(userName)) {
+async function createUser (userName) {
+  const hasUser = await UserDAO.hasUser(userName)
+  if (hasUser) {
```

Comment:
I find this variable (and the function) names a bit confusing. I would rather call it something like `doesUserAlreadyExist`, but again it's just style preferences.

I like assigning things to variables as you did, but just so you know, you could also write this as `if (await UserDAO.hasUser(userName)) {`